mne/preprocessing/_regress.py

```diff
@@ -191,8 +190,7 @@ def fit(self, inst):
 
         # Store relevant parameters in the object.
         self.coef_ = coef
-        all_picks = np.unique(np.hstack((picks, picks_artifact)))
-        self.info_ = pick_info(inst.info, all_picks)
+        self.info_ = inst.info.copy()
```

Comment:
@wmvanvliet this was buggy -- if `picks` and/or `picks_regression` were numerical and did not begin at 0 and stay contiguous, this could lead to incorrect channels being picked at the `apply` stage. Imagine you had 5 MEG channels, then 4 EEG, then 1 EOG, and your `picks` are `np.arange(5, 9)` (i.e., `EEG` but numerical), the `pick_info` of the picks+picks_artifact would reduce `info_` to only have 5 channels, but the numerical `.picks` attribute would be `np.arange(5, 9)`, which is clearly out of range.

We could probably get away with some `searchsorted` stuff, but it would be a bit complex and annoying to get right. It seems easy/okay enough just to have some extra `info['chs']` entries that won't ever be used...

Reply:
Huh. In my mind, `picks` would always be a list of str. Didn't I enforce that somewhere?

Reply:
Eventually some sanity check converts them to str but by that time the bug has already occurred. But in the object itself they were still stored as a single string (which generally worked) or as an ndarray of int (which didn't), depending on what was passed originally to the constructor

I agree storing the ordered channel names and using pick_channels(...,ordered=True) would be a good way to avoid the bug and avoid needing to keep am complete info